ludwig/modules/metric_modules.py

```diff
@@ -166,7 +167,7 @@ def __init__(self, name='r2_score'):
         self.add_state("sum_y_y_hat", default=torch.tensor(0, dtype=torch.float32))
         self.add_state("N", default=torch.tensor(0, dtype=torch.float32))
 
-    def update(self, y_hat, y):
+    def update(self, y_hat: Dict[str, torch.Tensor], y: torch.Tensor):
```

Comment:
this change would likely affect all the feature using this metric, namely the numerical ones. Should they be modified to pass the dictionary too?

Reply:
Currently the metric computation seems to be very inconsistent. A lot of metrics assume a dictionary of tensors as an argument while others assume a tensor. The update metrics function is also called from the `base_features` so consistency in argument types across metrics would be important

Reply:
Resolved offline